src/zope/password/password.py

```diff
@@ -450,13 +451,119 @@ def checkPassword(self, encoded_password, password):
         return encoded_password == self.encodePassword(password)[5:]
 
     def match(self, encoded_password):
-        if isinstance(encoded_password, unicode):
+        if isinstance(encoded_password, text_type):
             encoded_password = encoded_password.encode('ascii')
         return (
             encoded_password.startswith(b'{SHA}') or
             encoded_password.startswith(b'{SHA1}'))
 
 
+if bcrypt is not None:
```

Comment:
PasswordManagers are added to the [managers](/mgrbyte/zope.password/blob/244214631da14f268c85c309333ac517839dd086/src/zope/password/password.py#L568-L574) list at the bottom of the the password module,
so we wouldn't want to raise an exception in the `__init__ ` I don't believe.

Current there's a [guard](/mgrbyte/zope.password/blob/244214631da14f268c85c309333ac517839dd086/src/zope/password/password.py#L576-L577) around adding the `BCryptPasswordManager` to this list.

Perhaps I we should just remove the conditional definition of the class, and leave the conditional instantiation and append to the `managers` list in place?